sqlglot/dialects/exasol.py

```diff
@@ -0,0 +1,453 @@
+from __future__ import annotations
+import typing as t
+from sqlglot import exp, generator, parser, tokens, parse_one
+from sqlglot.dialects.dialect import (
+    Dialect,
+    binary_from_function,
+    build_date_delta,
+    rename_func,
+)
+from sqlglot.helper import seq_get
+from sqlglot.tokens import TokenType
+from sqlglot.generator import unsupported_args
+
+DATEΤΙΜΕ_DELTA = t.Union[
+    exp.DateAdd, exp.DateDiff, exp.DateSub, exp.TimestampSub, exp.TimestampAdd
+]
+
+
+
+
+class Tokenizer(tokens.Tokenizer):
+    IDENTIFIER_ESCAPES = ['"']
+    STRING_ESCAPES = ["'"]
+
+    KEYWORDS = {
+        **tokens.Tokenizer.KEYWORDS,
+        "YEAR": TokenType.YEAR,
+        "WITH LOCAL TIME ZONE": TokenType.TIME,
+        "TO": TokenType.COMMAND,
+        "HASHTYPE": TokenType.UUID,
+    }
+
+
+
+
+
+def _string_position_sql(self: Exasol.Generator, expression: exp.StrPosition) -> str:
+    """
+    Generate Exasol SQL for string position expressions.
+
+    Exasol supports:
+    - POSITION(substr IN string)
+    - LOCATE(substr, string, position)
+    - INSTR(string, substr, position, occurrence)
+
+    sqlglot parses all into `exp.StrPosition`, and this method emits the correct SQL form
+    based on which arguments are present.
+    """
+    this = expression.args.get("this")
+    substr = expression.args.get("substr")
+    position = expression.args.get("position")
+    occurrence = expression.args.get("occurrence")
+    # POSITION format
+    if position is None and occurrence is None:
+        return f"POSITION({self.sql(substr)} IN {self.sql(this)})"
+
+    position_sql = self.sql(position)
+    occurrence_sql = self.sql(occurrence)
+
+    # LOCATE with optional position
+    if occurrence is None:
+        if position_sql == "1":
+            return self.func("LOCATE", substr, this)
+        return self.func("LOCATE", substr, this, position)
+
+    # INSTR default simplification
+    if position_sql == "1" and occurrence_sql == "1":
+        return self.func("INSTR", this, substr)
+
+    # Full INSTR
+    return self.func("INSTR", this, substr, position, occurrence)
+
+
+
+class Exasol(Dialect):
+    ANNOTATORS = {
+        **Dialect.ANNOTATORS,
+    }
+
+    DATE_FORMAT = "'yyyy-MM-dd'"
+    DATEINT_FORMAT = "'yyyyMMdd'"
+    TIME_FORMAT = "'yyyy-MM-dd HH:mm:ss'"
+
+    TIME_MAPPING = {
+        # --- Year ---
+        "YYYY": "%Y",  # 4-digit year (e.g., 2025)
+        "YYY": "%Y",  # 3-digit year (often same as YYYY in practice for TO_CHAR)
+        "YY": "%y",  # Last 2 digits of year (e.g., 25)
+        "Y": "%y",  # Last digit of year
```

Comment:
These don't look right. For example, `YYYY` will roundtrip as `YYY`– aren't these two different?